app/main.py

```diff
@@ -1,2 +1,47 @@
+from typing import Hashable, Any
+
+
 class Dictionary:
-    pass
+    def __init__(self) -> None:
+        self.length = 0
+        self.capacity = 8
+        self.hash_table = [None] * self.capacity
+
+    def __setitem__(self, key: Hashable, value: Any) -> None:
+        index = hash(key) % self.capacity
+        if self.length >= round(self.capacity * 2 / 3) - 1:
+            self.length = 0
+            self.capacity *= 2
+            hash_table = self.hash_table[:]
+            self.hash_table = [None] * self.capacity
+            for node in hash_table:
+                if node:
+                    self[node[0]] = node[2]
+        if not self.hash_table[index]:
+            self.length += 1
+            self.hash_table[index] = [key, hash(key), value]
+        else:
+            while True:
+                if not self.hash_table[index]:
+                    self.length += 1
+                    self.hash_table[index] = [key, hash(key), value]
+                    break
+                if self.hash_table[index][0] == key:
+                    self.hash_table[index][2] = value
+                    break
+                index = index + 1
+                index = index % self.capacity
```

Comment:
Liudmyla meant that you can separate resizing into additional function to improve readability and not overloading __setitem__